stix_shifter/stix_transmission/src/modules/bigfix/bigfix_status_connector.py

```diff
@@ -51,39 +49,51 @@ def create_status_connection(self, search_id):
 
                     reporting_agents = int(response_dict.get('reportingAgents', '0'))
                     total_results = int(response_dict.get('totalResults', '0'))
-
-                    if self.DEFAULT_CLIENT_COUNT == client_count:
-                        return_obj['progress'] = 0
-                    else:
-                        progress = (reporting_agents / client_count) * 100
-                        progress_floor = math.floor(progress)
-                        return_obj['progress'] = progress_floor
-
+                    return_obj = self._get_progress_status(client_count, reporting_agents, return_obj)
                     if client_count <= reporting_agents:
                         return_obj['status'] = Status.COMPLETED.value
                         if total_results <= 0:
                             return_obj['status'] = Status.ERROR.value
-
-                    if return_obj['success'] == True and return_obj['status'] == Status.COMPLETED.value:
-                        if not self.get_succes_status(response_dict):
-                            return_obj['status'] = Status.ERROR.value
-                            ErrorResponder.fill_error(return_obj, response_dict, ['results', '+isFailure=True','~result'])
-
+                        break
+                    time_iter -= WAIT_PERIOD
+                    if time_iter < WAIT_PERIOD:
```

Comment:
isn't it safe to track the timestamp instead of counting wait period?

Reply:
@mdazam1942 
Can you please clarify provide more details on this query?